src/VisualStudio/Core/Def/SampleToolWindow/SampleToolboxUserControl.xaml.cs

```diff
@@ -0,0 +1,425 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.Editor;
+using Microsoft.VisualStudio.LanguageServer.Client;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.OLE.Interop;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.TextManager.Interop;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.VisualStudio.LanguageServices
+{
+    using Workspace = Microsoft.CodeAnalysis.Workspace;
+
+    /// <summary>
+    /// Interaction logic for SampleToolboxUserControl.xaml
+    /// </summary>
+    internal partial class SampleToolboxUserControl : UserControl, IOleCommandTarget
+    {
+        public SampleToolboxUserControl(Workspace workspace, IDocumentTrackingService documentTrackingService, ILanguageServiceBroker2 languageServiceBroker, IThreadingContext threadingContext)
+        {
+            InitializeComponent();
+            InitializeIfNeeded(workspace, documentTrackingService, languageServiceBroker, threadingContext);
+        }
+
+        private Workspace? workspace { get; set; }
+        private DocumentId? lastDocumentId { get; set; }
+
+        [MemberNotNullWhen(true, nameof(symbolsTreeItemsSource))]
+        private bool symbolTreeInitialized { get; set; }
+        private ObservableCollection<DocumentSymbolViewModel>? symbolsTreeItemsSource { get; set; }
+
+        private ITextSnapshot? snapshot { get; set; }
+        private IWpfTextView? textView { get; set; }
+
+        private void InitializeIfNeeded(Workspace workspace, IDocumentTrackingService documentTrackingService, ILanguageServiceBroker2 languageServiceBroker, IThreadingContext threadingContext)
+        {
+            this.workspace = workspace;
+            documentTrackingService.ActiveDocumentChanged += DocumentTrackingService_ActiveDocumentChanged;
+
+            this.textView = GetActiveTextView();
+            if (this.textView is not null)
+            {
+                this.textView.Caret.PositionChanged += FollowCursor;
+            }
+
+            searchBox.GotFocus += RemoveText;
```

Comment:
instead of setting these event handlers here you can just add them in the xaml file
```xml
<TextBox Grid.Row="1"
         Name="searchBox"
         Text="Search Document Outline"
         TextChanged="Search"
         VerticalContentAlignment="Center"
         GotFocus="RemoveText"
         LostFocus="AddText" />
```